include/envoy/router/router.h

```diff
@@ -502,6 +502,16 @@ class MetadataMatchCriteria {
   mergeMatchCriteria(const ProtobufWkt::Struct& metadata_matches) const PURE;
 };
 
+class TlsContextMatchCriteria;
+using TlsContextMatchCriteriaConstPtr = std::unique_ptr<const TlsContextMatchCriteria>;
```

Comment:
Nit: move this after the class definition below, no need to forward declare for this simple class.

Reply:
removed

Reply:
I still see the forward declaration above..

Reply:
Sorry, thought it was referring to same comment as tls_context_match_criteria_impl.h below
Fixed now.